### PR TITLE
test: Test constructor function with and without values

### PR DESCRIPTION
**Pull request by Jrigada**
# What :computer: 
* Add tests for constructor function in contracts with and without values

# Why :hand:
* Values in constructors were not being passed and transferred to the contract.

### CODE REVIEW

**Review by nbaztec — DISMISSED** [review]
Tests always good :D

File: crates/forge/tests/fixtures/zk/Constructor.s.sol [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        // Test constructor with smaller value
+        Bank bankSmallValue = new Bank{value: 0.1 ether}();
+        assert(bankSmallValue.balance() == 0.1 ether);
+        
+        vm.stopBroadcast();
+    }
+}
+
+contract Bank {
+    event Received(address sender, uint256 amount);
```
Comment (Karrq) [inline crates/forge/tests/fixtures/zk/Constructor.s.sol:27]: is this event used anywhere?

File: testdata/zk/Constructor.t.sol [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return address(this).balance;
+    }
+
+    constructor() payable {}
+
+    receive() external payable {}
+}
+
+contract ZkConstructorTest is DSTest {
+    Vm constant vm = Vm(HEVM_ADDRESS);
```
Comment (Karrq) [inline testdata/zk/Constructor.t.sol:19]: necessary?

File: testdata/zk/Constructor.t.sol [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.18;
+
+import "ds-test/test.sol";
+import "../cheats/Vm.sol";
+import "../default/logs/console.sol";
+
+contract Bank {
```
Comment (Karrq) [inline testdata/zk/Constructor.t.sol:8]: I'd reuse the `Bank` contract, either from it's on file here and then importing it in the script project or the opposite

**Review by Karrq — APPROVED** [review]

**Pull request merged**